src/Syncs/BloomFilterSync.cpp

```diff
@@ -0,0 +1,217 @@
+/* This code is part of the GenSync project developed at Boston University.  Please see the README for use and references. */
+
+//
+// Created by Anish Sinha on 12/4/2023.
+//
+
+#include <GenSync/Aux/Exceptions.h>
+#include <GenSync/Syncs/BloomFilterSync.h>
+#include <iostream>
+
+BloomFilterSync::BloomFilterSync(size_t expNumElems, size_t eltSize, int szMult, int nHash)
+{
+	this->expNumElems = expNumElems;
+	elementSize = eltSize;
+        myBloomFilter = BloomFilter::Builder().
+                        setSize(expNumElems*szMult).
+                        setNumHashes(nHash).
+                        build();
+}
+
+BloomFilterSync::BloomFilterSync(size_t expNumElems, size_t eltSize, float falsePosProb)
+{
+        this->expNumElems = expNumElems;
+        elementSize = eltSize;
+        myBloomFilter = BloomFilter::Builder().
+                        setNumExpElems(expNumElems).
+                        setFalsePosProb(falsePosProb).
+                        build();
+}
+
+BloomFilterSync::~BloomFilterSync() = default;
+
+bool BloomFilterSync::SyncClient(const shared_ptr<Communicant>& commSync, list<shared_ptr<DataObject>> &selfMinusOther, list<shared_ptr<DataObject>> &otherMinusSelf)
+{
+	Logger::gLog(Logger::METHOD, "Entering BloomFilterSync::SyncClient");
+
+        bool success = true;
+
+        // call parent method for bookkeeping
+        SyncMethod::SyncClient(commSync, selfMinusOther, otherMinusSelf);
+
+        // connect to server
+        mySyncStats.timerStart(SyncStats::IDLE_TIME);
+        commSync->commConnect();
+        mySyncStats.timerEnd(SyncStats::IDLE_TIME);
+
+        // Verify server and client Bloom Filters have matching size and numHashes otherwise fail
+        mySyncStats.timerStart(SyncStats::COMM_TIME);
+
+        int clientBFsize = myBloomFilter.getSize();
+        int clientNumHashes = myBloomFilter.getNumHashes();
+
+        int serverBFsize = commSync->commRecv_int();
+        int serverNumHashes = commSync->commRecv_int();
+
+        commSync->commSend(clientBFsize);
+        commSync->commSend(clientNumHashes);
+
+        if((clientBFsize != serverBFsize) || (clientNumHashes != serverNumHashes)) {
+            Logger::gLog(Logger::METHOD_DETAILS, "BloomFilter parameters do not match up between client and server!");
+            mySyncStats.timerEnd(SyncStats::COMM_TIME);
+            mySyncStats.increment(SyncStats::XMIT,commSync->getXmitBytes());
```

Comment:
Would it be easier to add a method for commSync that includes the timer that should be updated?  That way you won't have to constantly start and stop the timer?

Reply:
Yes, I agree. Since this would require changes to the Communicant class, would it be best to make a separate issue for this?

Reply:
yes.  Good idea.